integration-tests/cucumber-tests-execution/src/main/java/org/opensmartgridplatform/cucumber/execution/JUnitCommandLineParseResult.java

```diff
@@ -58,12 +69,12 @@ private void parseArgs(final String[] args) {
         for (int i = 0; i != args.length; ++i) {
             final String arg = args[i];
 
-            if (arg.equals("--")) {
+            if (DASH_DASH.equals(arg)) {
                 return this.copyArray(args, i + 1, args.length);
-            } else if (arg.startsWith("--")) {
-                if (arg.startsWith("--filter=") || arg.equals("--filter")) {
+            } else if (DASH_DASH.startsWith(arg)) {
+                if (DASH_DASH_FILTER.startsWith(arg) || DASH_DASH_FILTER.equals(arg)) {
                     String filterSpec;
-                    if (arg.equals("--filter")) {
+                    if (DASH_DASH_FILTER.equals(arg)) {
                         ++i;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor the code in order to not assign to this loop counter from within the loop body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AForLoopCounterChangedCheck)